mlx/transforms.cpp

```diff
@@ -19,6 +19,38 @@
 
 namespace mlx::core {
 
+namespace {
+
+static std::atomic<int> _tracing_counter = 0;
+
+// Create an InTracing object during tracing operations to signify to the rest
+// of the codebase that we are during tracing so evals should not throw away
+// the graph.
+struct InTracing {
+ public:
```

Comment:
nit if you care for it, struct defaults public.

Reply:
Yeah I usually omit it if there is not gonna be a private field. When there is I add it but I removed it :-)